adapters/spi/tomcat-adapter-spi/src/main/java/org/keycloak/adapters/tomcat/GenericPrincipalFactory.java

```diff
@@ -22,46 +22,58 @@
 
 import javax.security.auth.Subject;
 import java.security.Principal;
-import java.security.acl.Group;
 import java.util.ArrayList;
-import java.util.Collection;
 import java.util.Enumeration;
-import java.util.Iterator;
+import java.util.Collection;
 import java.util.List;
 import java.util.Set;
 
 /**
  * @author <a href="mailto:ungarida@gmail.com">Davide Ungari</a>
  * @version $Revision: 1 $
  */
-public abstract class GenericPrincipalFactory {
+public abstract class GenericPrincipalFactory implements PrincipleFactory {
 
+    @Override
     public GenericPrincipal createPrincipal(Realm realm, final Principal identity, final Set<String> roleSet) {
-        Subject subject = new Subject();
-        Set<Principal> principals = subject.getPrincipals();
+        final Subject subject = new Subject();
+        final Set<Principal> principals = subject.getPrincipals();
```

Comment:
Please don't change code unrelated to the fix at hand.
Adding `final` everywhere possible is a controversial style issue, many people consider it useless noise in the source code.

Reply:
these were from the original PR, but I removed them anyway.